extension-manifest-v3/scripts/download-engines/utils.js

```diff
@@ -82,6 +73,8 @@ export function setupStream(path) {
 
   return {
     write: (rule) => {
+      if (!rule) return;
```

Comment:
when does this occur? shouldn't we at least log when this happen?

Reply:
It moves the condition from the generator to the stream. If you look at the change in the PR, now we call `write` always, but the Safari util sometimes returns null, for example when the rule is not supported.